aggregator/src/aggregator.rs

```diff
@@ -3223,9 +3224,31 @@ impl VdafOps {
                                 Error::UnrecognizedCollectionJob(collection_job_id).into(),
                             )
                         })?;
-                    Q::acknowledge_collection(tx, task.id(), collection_job.batch_identifier())
-                        .await?;
-                    Ok(collection_job)
+
+                    let (batch_aggregations, _) = try_join!(
+                        Q::get_batch_aggregations_for_collect_identifier(
+                            tx,
+                            &task,
+                            collection_job.batch_identifier(),
+                            collection_job.aggregation_parameter()
+                        ),
+                        Q::acknowledge_collection(tx, task.id(), collection_job.batch_identifier()),
+                    )?;
+
+                    // Merge the intervals spanned by the constituent batch aggregations into the
+                    // interval spanned by the collection.
+                    let mut spanned_interval: Option<Interval> = None;
+                    for interval in batch_aggregations
+                        .iter()
+                        .map(BatchAggregation::<L, Q, A>::client_timestamp_interval)
+                    {
+                        match spanned_interval {
+                            Some(m) => spanned_interval = Some(m.merge(interval)?),
+                            None => spanned_interval = Some(*interval),
+                        }
+                    }
+
+                    Ok((collection_job, spanned_interval))
```

Comment:
We should round up the interval, either before returning it here or during the aggregation flow. The text says "... the interval's start and duration are both multiples of the task's time_precision parameter." Batch aggregations are initialized with the result of `Interval::from_time(timestamp)`, which has duration 1. (currently, the aggregation tests expect batch aggregations with duration 1 intervals, while the collection tests set up batch aggregations with `time_precision`-length intervals and expect the same in collections)